src/bank.rs

```diff
@@ -339,6 +339,16 @@ impl Bank {
         }
     }
 
+    /// Maps a tick height to a timestamp
+    fn tick_height_to_timestamp(last_ids: &LastIds, tick_height: u64) -> Option<u64> {
```

Comment:
is this ok, even with the limited depth of last_ids?

Reply:
This is called from get_finality_timestamp() which holds the last_ids lock so no new ticks should be added, and also filters out any ticks that are outside of the MAX_ENTRY_IDS window. It should be guaranteed then that this function is only called on ticks that are within the MAX_ENTRY_IDS window. There is the chance that the only tick that the validators have reached consensus on is outside of the window of valid ticks. This then causes get_last_supermajority_timestamp() to return Err(FinalityError::NoValidSupermajority), and submits a point to the finality metric (now() - last finality time) that increases the average finality time.